lib/cartopy/io/img_tiles.py

```diff
@@ -320,6 +320,57 @@ def _image_url(self, tile):
         return f'https://a.tile.openstreetmap.org/{z}/{x}/{y}.png'
 
 
+class StadiaMapsTiles(GoogleWTS):
+    """
+    Retrieves tiles from stadiamaps.com.
+
+    For a full reference on the styles available please see
+    https://docs.stadiamaps.com/themes/. A few of the specific styles
+    that are made available are ``alidade_smooth``, ``stamen_terrain`` and
+    ``osm_bright``.
+
+    Using the Stadia Maps API requires including an attribution. Please see
+    https://docs.stadiamaps.com/attribution/ for details.
+
+    For most styles that means including the following attribution:
+
+    `© Stadia Maps <https://www.stadiamaps.com/>`_
+    `© OpenMapTiles <https://openmaptiles.org/>`_
+    `© OpenStreetMap contributors <https://www.openstreetmap.org/about/>`_
+
+    with Stamen styles *additionally* requiring the following attribution:
+
+    `© Stamen Design <https://stamen.com/>`_
+
+    Parameters
+    ----------
+    apikey : str, required
+        The authentication key provided by Stadia Maps to query their APIs
+    style : str, optional
+        Name of the desired style. Defaults to ``alidade_smooth``.
+        See https://docs.stadiamaps.com/themes/ for a full list of styles.
+    cache : bool or str, optional
+        If True, the default cache directory is used. If False, no cache is
+        used. If a string, the string is used as the path to the cache.
+    """
+
+    def __init__(self, apikey, style='alidade_smooth', cache=False):
+        super().__init__(cache=cache)
+        self.apikey = apikey
+        if style in ("alidade_satellite", "stamen_watercolor"):
+            # Known styles that have the jpg extension
+            self.extension = "jpg"
+        else:
+            self.extension = "png"
+        self.style = style
+
+    def _image_url(self, tile):
+        x, y, z = tile
+        return ("http://tiles.stadiamaps.com/tiles/"
+                f"{self.style}/{z}/{x}/{y}.{self.extension}"
```

Comment:
Dumb question more than anything as I regrettably haven't used CartoPy myself, but are @2x (aka retina or 512 tiles) useful? If so, we _do_ support those on all styles except for Stamen Watercolor by adding `@2x` right before the extension.

Reply:
👍 It is easy enough to add, so I added an extension parameter as well now and it looks good locally to me.